tests/test_odl.py

```diff
@@ -2090,3 +2093,135 @@ def test_odl_reaper_removes_expired_licenses(self):
+        [imported_pool] = imported_pools
+
+        # 1.3. Ensure that the two valid licenses were imported
+        assert len(imported_pool.licenses) == 2
+
+        # 1.4 Make sure that 20 licenses are marked as owned (10 from each valid license)
+        assert imported_pool.licenses_owned == self.LICENSES_OWNED * 2
+        assert imported_pool.licenses_available == self.LICENSES_AVAILABLE * 2
+
+    @freeze_time("2021-01-01T00:00:00+00:00")
+    def test_odl_reaper_removes_expired_licenses(self):
+        """Ensure ODLExpiredItemsReaper removes expired licenses."""
+        # 1.1. Import the test feed with an ODL licenses that are still valid.
+        license_expiration_dates = [
+            datetime_helpers.utc_now() + datetime.timedelta(days=1),
+            datetime_helpers.utc_now() + datetime.timedelta(days=60),
+            datetime_helpers.utc_now() + datetime.timedelta(days=365)
+        ]
+        imported_editions, imported_pools, imported_works = self._import_test_feed(license_expiration_dates)
+
+        # Commit to expire the SQLAlchemy cache.
+        self._db.commit()
+
+        # 1.2. Ensure that there is a license pool with available license.
+        assert len(imported_pools) == 1
+
+        [imported_pool] = imported_pools
+        assert imported_pool.licenses_owned == 3 * self.LICENSES_OWNED
+        assert imported_pool.licenses_available == 3 * self.LICENSES_AVAILABLE
+        assert len(imported_pool.licenses) == 3
+
+        [license1, license2, license3] = imported_pool.licenses
+        assert license1.expires == license_expiration_dates[0]
+        assert license2.expires == license_expiration_dates[1]
+        assert license3.expires == license_expiration_dates[2]
+
+        # 2.1. Run ODLExpiredItemsReaper. This time nothing should happen since the license is still valid.
+        script = RunCollectionMonitorScript(self.ODL_REAPER_CLASS, _db=self._db, cmd_args=["Test ODL Collection"])
+        script.run()
+
+        # Commit to expire the SQLAlchemy cache.
+        self._db.commit()
+
+        # 2.2. Ensure that availability of the license pool didn't change.
+        assert imported_pool.licenses_owned == 3 * self.LICENSES_OWNED
+        assert imported_pool.licenses_available == 3 * self.LICENSES_AVAILABLE
+        assert len(imported_pool.licenses) == 3
+
+        # 3. Expire the license.
+        license1.expires = datetime_helpers.utc_now() - datetime.timedelta(days=1)
+
+        # 3.1. Run ODLExpiredItemsReaper again. This time it should remove the expired license.
+        script.run()
+
+        # Commit to expire the SQLAlchemy cache.
+        self._db.commit()
+
+        # 3.2. Ensure that availability of the license pool was updated
+        assert imported_pool.licenses_owned == 2 * self.LICENSES_OWNED
+        assert imported_pool.licenses_available == 2 * self.LICENSES_AVAILABLE
+
+        # 4.1. Run ODLExpiredItemsReaper again to ensure that number of licenses won't become negative.
```

Comment:
Is "negative" what you mean here or do you mean that license count should not decrement?